Source: Trond A Ekseth <troeks@gmail.com>
Upstream: No
Reason: Binding to 0.0.0.0 creates access violations under sydbox.

diff -ru gnutls-2.8.5/tests/anonself.c gnutls-2.8.5-sydbox/tests/anonself.c
--- gnutls-2.8.5/tests/anonself.c	2010-01-16 19:57:38.861160331 +0100
+++ gnutls-2.8.5-sydbox/tests/anonself.c	2010-01-16 20:12:44.426944474 +0100
@@ -213,7 +213,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/dhepskself.c gnutls-2.8.5-sydbox/tests/dhepskself.c
--- gnutls-2.8.5/tests/dhepskself.c	2010-01-16 19:57:38.861160331 +0100
+++ gnutls-2.8.5-sydbox/tests/dhepskself.c	2010-01-16 20:12:44.433095005 +0100
@@ -217,7 +217,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/openpgpself.c gnutls-2.8.5-sydbox/tests/openpgpself.c
--- gnutls-2.8.5/tests/openpgpself.c	2010-01-16 19:57:38.861160331 +0100
+++ gnutls-2.8.5-sydbox/tests/openpgpself.c	2010-01-16 20:12:44.439954388 +0100
@@ -349,7 +349,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/oprfi.c gnutls-2.8.5-sydbox/tests/oprfi.c
--- gnutls-2.8.5/tests/oprfi.c	2010-01-16 19:57:38.862160667 +0100
+++ gnutls-2.8.5-sydbox/tests/oprfi.c	2010-01-16 20:12:44.443850652 +0100
@@ -289,7 +289,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/pskself.c gnutls-2.8.5-sydbox/tests/pskself.c
--- gnutls-2.8.5/tests/pskself.c	2010-01-16 19:57:38.862160667 +0100
+++ gnutls-2.8.5-sydbox/tests/pskself.c	2010-01-16 20:12:44.446098831 +0100
@@ -203,7 +203,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/resume.c gnutls-2.8.5-sydbox/tests/resume.c
--- gnutls-2.8.5/tests/resume.c	2010-01-16 19:57:38.862160667 +0100
+++ gnutls-2.8.5-sydbox/tests/resume.c	2010-01-16 20:12:44.447098601 +0100
@@ -255,7 +255,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/tlsia.c gnutls-2.8.5-sydbox/tests/tlsia.c
--- gnutls-2.8.5/tests/tlsia.c	2010-01-16 19:57:38.862160667 +0100
+++ gnutls-2.8.5-sydbox/tests/tlsia.c	2010-01-16 20:12:44.449098694 +0100
@@ -375,7 +375,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/x509dn.c gnutls-2.8.5-sydbox/tests/x509dn.c
--- gnutls-2.8.5/tests/x509dn.c	2010-01-16 19:57:38.862160667 +0100
+++ gnutls-2.8.5-sydbox/tests/x509dn.c	2010-01-16 20:12:44.450848494 +0100
@@ -382,7 +382,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/x509self.c gnutls-2.8.5-sydbox/tests/x509self.c
--- gnutls-2.8.5/tests/x509self.c	2010-01-16 19:57:38.862967296 +0100
+++ gnutls-2.8.5-sydbox/tests/x509self.c	2010-01-16 20:12:44.452098669 +0100
@@ -346,7 +346,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
diff -ru gnutls-2.8.5/tests/x509signself.c gnutls-2.8.5-sydbox/tests/x509signself.c
--- gnutls-2.8.5/tests/x509signself.c	2010-01-16 19:57:38.862967296 +0100
+++ gnutls-2.8.5-sydbox/tests/x509signself.c	2010-01-16 20:12:44.454098765 +0100
@@ -347,7 +347,7 @@
 
   memset (&sa_serv, '\0', sizeof (sa_serv));
   sa_serv.sin_family = AF_INET;
-  sa_serv.sin_addr.s_addr = INADDR_ANY;
+  sa_serv.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
   sa_serv.sin_port = htons (PORT);	/* Server Port number */
 
   setsockopt (listen_sd, SOL_SOCKET, SO_REUSEADDR, (void *) &optval, sizeof (int));
